go/interpreter/lfvm/memory_test.go

```diff
@@ -185,3 +192,272 @@ func TestMemory_expandMemory_expandsMemoryOnlyWhenNeeded(t *testing.T) {
 		})
 	}
 }
+
+func TestMemory_getSlice_ErrorCases(t *testing.T) {
+	c := getEmptyContext()
+	c.gas = 0
```

Comment:
As I just realized, `getEmptyContext()` does not return an empty context. It returns a context with some `gas`. This is very misleading. Why not simpler
```suggestion
	c := &context{}
```